lib/wallet.js

```diff
@@ -315,6 +315,144 @@ class WalletClient extends Client {
     return this.get(`/wallet/${id}/coin/${hash}/${index}`);
   }
 
+  /**
+   * Get name state for the given name.
+   * {@see hsd/lib/covenants/namestate/NameState}
+   * @param {String} id
+   * @param {String} name
+   * @returns {Promise}
+   */
+
+  getName(id, name) {
+    return this.get(`/wallet/${id}/name/${name}`);
+  }
+
+  /**
+   * Get name state for all names
+   * that the wallet is managing.
+   * {@see hsd/lib/covenants/namestate/NameState}
+   * @param {String} id
+   * @returns {Promise}
+   */
+
+  getNames(id) {
```

Comment:
This could take `options` as well for future proofing, but the endpoint currently does not support `options`. This would allow for pagination or filtering in the future.

Reply:
I think we could add options later